### DISCUSSION TITLE
Lambda Expression

### DISCUSSION THREAD

**Discussion by chauthanhdat** [ChebanovDD/MatchSweets#5] [Q&A] [answered]
Hi, thanks for your project.
I have a little question about the lamdba expression. At the beginning of some class I found you using "=>" to assign value to variable instead of "=". Example:
```

[SerializeField] private AppContext _appContext;
[SerializeField] private InteractableDropdown _iconsSetDropdown;
[SerializeField] private InteractableDropdown _fillStrategyDropdown;
[SerializeField] private InteractableButton _startGameButton;

public int SelectedIconsSetIndex => _iconsSetDropdown.SelectedIndex;
public int SelectedFillStrategyIndex => _fillStrategyDropdown.SelectedIndex;

```
Can you explain why you use "=>" instead of "="?
Thank you.

**Comment by ChebanovDD** [comment, marked as answer]
Hi there! These are different things.

The `=` is an [assignment operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/assignment-operator).
The `=>` is an [expression-bodied member](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members).

Expression-bodied read-only property (syntactic sugar):
```csharp
public int SelectedFillStrategyIndex => _fillStrategyDropdown.SelectedIndex;
```

How the compiler sees it:
```csharp
public int SelectedFillStrategyIndex
{
    get { return _fillStrategyDropdown.SelectedIndex; }
}
```
Expression body definition is just a way to implement a read-only property in a very concise, readable form.
